auth0/src/main/java/com/auth0/android/management/UsersAPIClient.kt

```diff
@@ -186,8 +187,12 @@ public class UsersAPIClient @VisibleForTesting(otherwise = VisibleForTesting.PRI
         val userProfileAdapter: JsonAdapter<UserProfile> = GsonAdapter(
             UserProfile::class.java, gson
         )
-        return factory.patch(url.toString(), userProfileAdapter)
-            .addParameter(USER_METADATA_KEY, gson.toJson(userMetadata))
+        val patch = factory.patch(
+            url.toString(),
+            userProfileAdapter
+        ) as BaseRequest<UserProfile, ManagementException>
```

Comment:
"unsafe cast": would throw if the cast cannot be made. But we know the type our RequestFactor uses internally.